### ISSUE TITLE
Whitespace stripped from markdown despite settings, again

### ISSUE THREAD

**Issue by thosch66** [atom/whitespace#185]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Whitespace is stripping trailing whitespaces in markdown files despited the settings.

### Steps to Reproduce

1. Edit a README.md for a project
2. Put two whitespaces at end of line to indicate a line break.
3. Save the file.

**Expected behavior:** The whitespace at the end of the line are ignored and and not deleted.

**Actual behavior:** The whitespaces are deleted. (The trailing whitespaces in the current line are ignored.) If the option "Remove Trailing Whitespace" or the package Whitespace is deactivated, the whitespaces remain.

**Reproduces how often:** 100.0000000%

### Versions

Atom    : 1.33.0
Electron: 2.0.11
Chrome  : 61.0.3163.100
Node    : 8.9.3

Whitespace: 0.37.7

OS: Mac OS Mohave 10.14.1

### Additional Information

I can **not** reproduce the issue running Atom on Debian (Bunsen Labs).

**Comment by rsese** [comment]
Thanks for the report - I'm unable to reproduce with 1.33.0 on macOS 10.12.6.  Can you share your settings for `Ignore Whitespace On Current Line` and `Keep Markdown Line Break Whitespace` as well?

**Comment by thosch66** [comment]
> Can you share your settings for Ignore Whitespace On Current Line and Keep Markdown Line Break Whitespace as well?

Both are activated.

**Comment by thosch66** [comment]
I have no access to the machine for the next days.  Stay tuned…

**Comment by thosch66** [comment]
> After resetting to the factory settings, the problem was solved until I reinstalled the package [language-markdown](https://atom.io/packages/language-markdown). (On the linux machine is not installed…)
> 
> I will report a bug for language-markdown.

Or perhaps it's the combination of `language-markdown` and `whitespace`: `language-markdown` changes an attribute and `whitespace` can not recognize the source as markdown?

**Comment by burodepeper** [comment]
@rsese It's not as much that `language-markdown` doesn't respect the setting, but `whitespace` should now also work within the `text.md` scope. I can't remember to what extend I've tested it. Maybe this can be added to this package's spec?

**Comment by thosch66** [comment]
> @rsese It's not as much that `language-markdown` doesn't respect the setting, but `whitespace` should now also work within the `text.md` scope. I can't remember to what extend I've tested it. Maybe this can be added to this package's spec?

This supports my assumtion, that there is a problem with combination of `whitespace` and  `language-markdown`.

After adding

    ".text.md":
        whitespace:
        removeTrailingWhitespace: false

to my `config.cson`, the problem is solved.

I agree with @burodepeper that `whitespace` should also respect the scope `text.md` as markdown file.

**Comment by burodepeper** [comment]
I found an issue in my PR in #177 and I've submitted a new PR #186 to fix it. I wrote silly Javascript, I take the blame.

**Comment by rsese** [comment]
Just circling back to this, thanks @burodepeper!   I'll edit this issue body today so it's clear the problem is that the whitespace package isn't working with the `text.md` scope as it was meant to after #177.

**Comment by ivanpu** [comment]
Um... I made a PR for this, but it was attached to the wrong repository... is it possible to migrate it, or should I create a new one from scratch?

**Comment by burodepeper** [comment]
I haven't had time to add tests to my previous PR. If you find time to do it for yours, feel free to do so, then I'll remove my PR.

**Comment by ivanpu** [comment]
I don't know how to add tests, and couldn't find how to test manually on my machine (couldn't find this package files within atom directories). Is there some documentation for this?

**Comment by rsese** [comment]
Our documentation on writing tests is here (it covers running tests as well):

https://flight-manual.atom.io/hacking-atom/sections/writing-specs/

If you need help writing automated tests for Atom, check out [Discuss, the official Atom and Electron message board](https://discuss.atom.io) or [join the Atom Slack team](http://atom-slack.herokuapp.com/). There are a bunch of helpful community members that should be willing to point you in the right direction (I'm not an engineer so can't help you myself unfortunately).

**Comment by thosch66** [comment]
@rsese 

> or [join the Atom Slack team](http://atom-slack.herokuapp.com/).

This link is down for months...

**Comment by rsese** [comment]
I messaged you on the message board @thosch66, we can talk more there.

**Comment by thosch66** [comment]
@rsese Thanks. I found the thread in the message board *after* posted here.

**Comment by ivanpu** [comment]
@rsese Thanks, will look at it, hopefully soon - I'm too busy these days with work.
